sunkit_image/granule.py

```diff
@@ -0,0 +1,331 @@
+    segmented_image = np.uint8(median_filtered > threshold)
+
+    # fix the extra IGM bits in the middle of granules
+    seg_im_fixed = trim_intergranules(segmented_image, mark=mark_dim_centers)
+
+    # mark faculae and get final granule and facule count
+    seg_im_markfac, fac_cnt, gran_cnt = mark_faculae(seg_im_fixed, data, res)
+    logging.info("Segmentation has identified " + str(gran_cnt) + " granules and " + str(fac_cnt) + "faculae")
+
+    # convert segmentated image back into SunPy map with original header
+    segmented_map = sunpy.map.Map(seg_im_markfac, header)
+
+    return segmented_map
+
+
+def get_threshold(data, method):
+    """
+    Get the threshold value using given skimage segmentation type.
+
+    Parameters
+    ----------
+    data : `numpy.ndarray`
+        Data to threshold.
+    method : `str`
+        Skimage thresholding method - options are 'otsu', 'li', 'isodata',
+        'mean', 'minimum', 'yen', 'triangle'.
+
+    Returns
+    -------
+    threshold `float`:
+        Threshold value.
+    """
+
+    if not isinstance(data, np.ndarray):
+        raise ValueError("Input data must be an array.")
+
+    methods = ["otsu", "li", "isodata", "mean", "minimum", "yen", "triangle"]
+    if method not in methods:
+        raise ValueError("Method must be one of: " + str(methods))
+    if method == "otsu":
+        threshold = skimage.filters.threshold_otsu(data)
+    elif method == "li":
+        threshold = skimage.filters.threshold_li(data)
+    elif method == "yen":
+        threshold = skimage.filters.threshold_yen(data)
+    elif method == "mean":
+        threshold = skimage.filters.threshold_mean(data)
+    elif method == "minimum":
+        threshold = skimage.filters.threshold_minimum(data)
+    elif method == "triangle":
+        threshold = skimage.filters.threshold_triangle(data)
+    elif method == "isodata":
+        threshold = skimage.filters.threshold_isodata(data)
+
+    return threshold
+
+
+def trim_intergranules(segmented_image, mark=False):
+    """
+    Remove the erronous idenfication of intergranule material in the middle of
+    granules that pure threshold segmentation produces.
+
+    Parameters
+    ----------
+    segmented_image : `numpy.ndarray`
+        The segmented image containing incorrect extra intergranules.
+    mark : `bool`
+        If False, remove erronous intergranules. If True, mark them as 0.5
+        instead (for later examination).
+
+
+    Returns
+    -------
+    segmented_image_fixed : `numpy.ndarray`
+        The segmented image without incorrect extra intergranules.
+    """
+
+    if len(np.unique(segmented_image)) > 2:
+        raise ValueError("segmented_image must have only values of 1 and 0")
+
+    segmented_image_fixed = np.copy(segmented_image).astype(float)
+    labeled_seg = skimage.measure.label(segmented_image + 1, connectivity=2)
+    values = np.unique(labeled_seg)
+    # find value of the large continuous 0-valued region

```

Comment:
```suggestion
    # Find value of the large continuous 0-valued region.

```

Reply:
I am assuming all comments should be capitalized. Should they all have periods as well? I see that some of your changes do and some do not.

Reply:
Ideally yes, I would like the comments to look nice.

If I missed places, that was a mistake on my part.